dhis-2/dhis-api/src/main/java/org/hisp/dhis/trackedentity/TrackedEntityInstanceQueryParams.java

```diff
@@ -403,9 +403,7 @@ public TrackedEntityInstanceQueryParams addAttributes( List<QueryItem> attrs )
 
     public boolean hasFilterForEvents()
     {
-        return this.getAssignedUserQueryParam().hasAssignedUsers()
-            || AssignedUserSelectionMode.ANY == this.getAssignedUserQueryParam().getMode()
-            || AssignedUserSelectionMode.NONE == this.getAssignedUserQueryParam().getMode()
+        return this.getAssignedUserQueryParam().getMode() != AssignedUserSelectionMode.ALL
```

Comment:
This seems correct to me. It occurs to me the previous solution would be open to miss data; I think you might have solved a potential bug as well :)